### PR TITLE
Reorganize update section

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | n/a
| Versions      | all

Merges the 3.0 update with regular update guide.

- Both Update and Migration are now placed in one top-level navigation category.
- Misc steps at the end of the procedure merged into one "Finish the update" step.
- Replaced variable with product name spelled out. This change goes to all versions, where variables may be different/non existent and that frankly causes more problems than it does good.
- Renamed some pages for clarify and better searchability

(Note for reviewers:
The 4_x steps for updating 3.0 code were moved only, not modified.
The 5_ steps were also mostly moved with small text clarifications, but GH got confused and marked some pages as new.)

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -541,49 +541,39 @@ nav:
...
-        - '6. Dump assets': 'updating/6_dump_assets.md'
-        - '7. Commit, test and merge': 'updating/7_commit_test_merge.md'
+    - Update and migration:
+        - Updating Ibexa DXP:
+            - 'Updating to v3.3': 'updating/updating_to_3.3.md'
+            - 'Updating Ibexa DXP': 'updating/updating.md'
+            - '1. Check out a version': 'updating/1_check_out_version.md'
+            - '2. Resolve conflicts': 'updating/2_merge_composer.md'
+            - '3. Update the app': 'updating/3_update_app.md'
+            - '4. Upgrade the code':
```
Comment (mnocon) [inline mkdocs.yml:551]: Is there a reason why some things are updated` (app, database) and some are upgraded (code)?

File: docs/updating/2_merge_composer.md [hunk 2]
```diff
@@ -1,36 +1,59 @@
...
 
-## Manual merging
+## 2.1. Resolve conflicts in `composer.lock`
 
-!!! note "Sorted packages since 2.4 changes"
+Ignore the conflicts in `composer.lock`, because this file is regenerated when you execute `composer update` later.
+It is easiest to check out the version of the file from the tag and add it to the changes.
 
-    Because the 2.4 packages in `composer.json` are sorted, there will be more conflicts when updating to 2.4, but far fewer conflicts in the future. This is controlled by [sort-packages](https://getcomposer.org/doc/06-config.md#sort-packages) config in `composer.json`.
+If you get a lot of conflicts (for example, on the `doc` folder),
```
Comment (mnocon) [inline docs/updating/2_merge_composer.md:8]: It feels like this part deserves its own "chapter", it's not about composer.lock

File: mkdocs.yml (hunk 1 above)
Comment (DominikaK) [reply mkdocs.yml:551]: The idea was that update is smaller and more automatic, while upgrade is larger, requiring more hands-on work (as with upgrading custom code to v3), but I'm not married to the idea, let's see what the rest of the doc team has to say about it.

File: docs/updating/2_merge_composer.md (hunk 2 above)
Comment (DominikaK) [reply docs/updating/2_merge_composer.md:8]: So maybe this under general "Resolve conflicts" heading, and specifics about `composer.json` in a separate section below, like they are now?

File: docs/updating/5_update_2.3.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Apply the following database update script:
+
+``` bash
+mysql -u <username> -p <password> <database_name> < vendor/ezsystems/ezpublish-kernel/data/update/mysql/dbupdate-7.2.0-to-7.3.0.sql
+```
+
+## Changes to timestamp
+
+A new timestamp column has been added in order to keep track of when items were trashed, this is exposed in the API but not yet in UI.
```
Comment (mnocon) [inline docs/updating/5_update_2.3.md:15]: > this is exposed in the API but not yet in UI.

not sure if this detail is important here (or if it's still valid).

File: docs/updating/6_finish_the_update.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+yarn install
+yarn encore prod
+```
+
+## 6.3. Commit, test and merge
+
+Once all the conflicts have been resolved, and `composer.lock` updated, you can commit the merge.
+
+Note that you may or may not keep `composer.lock`, depending on your version management workflow.
+If you do not wish to keep it, run `git reset HEAD <file>` to remove it from the changes.
```
Comment (mnocon) [inline docs/updating/6_finish_the_update.md:37]: It seems that using `<file>` is not needed here, we could probably use `composer.lock` directly?

File: docs/updating/6_finish_the_update.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+git checkout master
+git merge <branch_name>
+```
+
+!!! note "Insecure password hashes"
+
+    To ensure that no users have unsupported, insecure password hashes, run the following command:
+    
+    ``` bash
+    php bin/console ezplatform:user:validate-password-hashes
```
Comment (mnocon) [inline docs/updating/6_finish_the_update.md:52]: For Ibexa DXP this command will work, but with a deprecation notice - the best version would be `php bin/console ibexa:user:validate-password-hashes`

And if I'm not wrong it's available only for 3.1+ (https://github.com/ezsystems/ezplatform-kernel/pull/99), so does not apply for v1 and v2.

**Review by mnocon — COMMENTED** [review]
It's hard to review that many files without being able to see how it will look live, but I think it will all make sense 😄

**Comment by mnocon** [comment]
1) Zrzut ekranu 2021-03-30 o 13 38 12

> If your application consists of several packages that are placed in locations other than the src/ folder, apply the suggestions from the upgrade documentation to all the packages before you run composer update.

I don't know what the "upgrade documentation" is, is it the "Upgrade the app" part? I think a link here would be useful.

2) 
Zrzut ekranu 2021-03-30 o 13 39 43

> Perform these steps only if you are updating to version v3.0. If not, move on to the 5. Update the database step.

The "Update the database" link is a dead one.

File: docs/updating/6_finish_the_update.md [hunk 6]
```diff
@@ -49,7 +49,7 @@ git merge <branch_name>
     To ensure that no users have unsupported, insecure password hashes, run the following command:
     
     ``` bash
-    php bin/console ezplatform:user:validate-password-hashes
+    php bin/console ibexa:user:validate-password-hashes
```
Comment (mnocon) [inline docs/updating/6_finish_the_update.md:52]: If I understand correctly this page is meant for users using all possible versions:
```
# In v1 and v2:
php bin/console assetic:dump -e prod
# In v2 and v3:
yarn install
yarn encore prod
```

So maybe we should mention that for version 3.2 you should run:
`php bin/console ezplatform:user:validate-password-hashes`
For 3.3:
`php bin/console ibexa:user:validate-password-hashes`
And for v1 and v2 it's not available.

**Review by mnocon — APPROVED** [review]

**Review by juskora — CHANGES_REQUESTED** [review]

**Review by juskora — APPROVED** [review]

**Pull request merged**